monitors.go

```diff
@@ -41,35 +42,18 @@ func (m *Monitor) JobUpdate(updateKey aurora.JobUpdateKey, interval int, timeout
 		Limit: 1,
 	}
 
-	defaultBackoff := m.Client.RealisConfig().backoff
-	duration := defaultBackoff.Duration //defaultBackoff.Duration
-	var err error
+	var cliErr error
 	var respDetail *aurora.Response
 
-	for i := 0; i*interval <= timeout; i++ {
-		for step := 0; step < defaultBackoff.Steps; step++ {
-			if step != 0 {
-				adjusted := duration
-				if defaultBackoff.Jitter > 0.0 {
-					adjusted = Jitter(duration, defaultBackoff.Jitter)
-				}
-				fmt.Println(" sleeping for: ", adjusted)
-				time.Sleep(adjusted)
-				duration = time.Duration(float64(duration) * defaultBackoff.Factor)
-			}
-			if respDetail, err = m.Client.JobUpdateDetails(updateQ); err == nil {
-				break
-			}
-			err1 := m.Client.ReestablishConn()
-			if err1 != nil {
-				fmt.Println("error in ReestablishConn: ", err1)
-			}
+	retryErr := ExponentialBackoff(*m.Client.RealisConfig().backoff, func() (bool, error) {
+		respDetail, cliErr = CheckAndRetryConn(m.Client, func() (*aurora.Response, error) {
+			return m.Client.JobUpdateDetails(updateQ)
+		})
+		if cliErr == RetryConnErr {
```

Comment:
This won't work the way we hope it would. https://play.golang.org/p/ONQlXcXDzl

We can compare cliErr.Error() and RetryConnErr.Error() for now until we get our Error story sorted out (https://stackoverflow.com/questions/39121172/how-to-compare-golang-error-objects )

Reply:
Will change this